src/libstd/sys/hermit/net.rs

```diff
@@ -17,14 +20,31 @@ pub fn init() -> io::Result<()> {
     Ok(())
 }
 
-pub struct TcpStream(abi::Handle);
+#[derive(Debug, Clone)]
+pub struct Socket(abi::Handle);
+
+impl AsInner<abi::Handle> for Socket {
+    fn as_inner(&self) -> &abi::Handle {
+        &self.0
+    }
+}
+
+impl Drop for Socket {
+    fn drop(&mut self) {
+        let _ = abi::tcpstream::close(self.0);
+    }
+}
+
+
+#[derive(Clone)]
+pub struct TcpStream(Arc<Socket>);
```

Comment:
why we need the `Arc` here?

Reply:
I want to close it once. Hm, should I use a reference to the handle?

Reply:
I see. Could you add a comment in the code explaining the use of Arc?

Reply:
Yeah, good idea!